### PR TITLE
refacto(*): remove everything about default workspace

### LINKED ISSUE #9033: Impersonation in multi-workspace context
Currently it's only possible to impersonate someone  on their default workspace.
We need to rethink impersonation in the context of the new structure that attributes one domain to each workspace.

### PR DESCRIPTION
**Pull request by AMoreaux**
## Summary
- [x] Remove defaultWorkspace in user
- [x] Remove all occurrence of defaultWorkspace and defaultWorkspaceId
- [x] Improve activate workspace flow
- [x] Improve security on social login
- [x] Add `ImpersonateGuard`
- [x] Allow to use impersonation with couple `User/Workspace`
- [x] Prevent unexpected reload on activate workspace
- [x] Scope login token with workspaceId 

Fix https://github.com/twentyhq/twenty/issues/9033#event-15714863042

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/admin-panel/admin-panel.resolver.ts [hunk 1]
```diff
@@ -6,49 +6,41 @@ import { ImpersonateInput } from 'src/engine/core-modules/admin-panel/dtos/imper
...
+  @UseGuards(WorkspaceAuthGuard, UserAuthGuard, ImpersonateGuard)
+  @Mutation(() => ImpersonateOutput)
   async impersonate(
-    @Args() impersonateInput: ImpersonateInput,
-    @AuthUser() user: User,
-  ): Promise<Verify> {
-    return await this.adminService.impersonate(impersonateInput.userId, user);
+    @Args() { workspaceId, userId }: ImpersonateInput,
+  ): Promise<ImpersonateOutput> {
+    return await this.adminService.impersonate(userId, workspaceId);
```
Comment (FelixMalfait) [inline packages/twenty-server/src/engine/core-modules/admin-panel/admin-panel.resolver.ts:25]: Should we check workspace.allowImpersonation on the backend? I didn't see it checked on the backend, maybe I missed it

**Review by FelixMalfait — APPROVED** [review]
LGTM! Tested and couldn't find any issue (or security issue / besides the question I asked on impersonation which is still open). Still it's something we should retest in pre-prod before the next release


(+ tests broken)

File: packages/twenty-server/src/engine/core-modules/admin-panel/admin-panel.resolver.ts (hunk 1 above)
Comment (AMoreaux) [reply packages/twenty-server/src/engine/core-modules/admin-panel/admin-panel.resolver.ts:25]: the impersonation of the workspace is checked in the SQL query 
```
     where: {
        id: userId,
        workspaces: {
          workspaceId,
          workspace: {
            allowImpersonation: true,
          },
        },
      },
```

Comment (FelixMalfait) [reply packages/twenty-server/src/engine/core-modules/admin-panel/admin-panel.resolver.ts:25]: I see! Thanks

**Pull request merged**